### PR TITLE
Adjust 'heroku releases' output to fit terminal width

### PR DESCRIPTION
**Pull request by HaraldNordgren**
This is a two-part fix:

- ~~First part is the fix in https://github.com/heroku/heroku-cli-util/pull/114 where I create a way to pass an options.optimizeKey to `cli.table` which will calculate a optimizationWidth consisting of the width of all other columns + column separators. And then pass this as a third paramater to the formatter for the optimization column.~~

- First part is this PR which expands the `descriptionWithStatus` formatter to use a calculated optimizationWidth to figure out exactly how wide the description column should be to fit the terminal width. I remove the arbitrary width 60 from the calculation and replaced it with a dynamic value. Special care is taken to use the status width _before_ appending the ANSI color tags instead of `sc` like the current code does.

- Second part is then to update https://github.com/heroku/cli to use this new code. I tested the combinations continuously with a real project during the development and it seems to work as intended. This fixes https://github.com/heroku/cli/issues/750.

The length calculation was always off by 2 length units. Displaying it like this for terminal width 80

```
$ heroku releases -a some-app
=== some-app Releases - Current: v446
v446  Deploy 11111111        another.name@email.se       2018/03/07 13:54:13 +01
00
v445  Set SOME_FEATUREFLA…   name@email.se               2018/03/05 08:09:05 +01
00
v444  Deploy 11111111        another.name@email.se       2018/03/01 10:23:18 +01
00
v443  Set SOME_FEATUREFLA…   another.name@email.se       2018/03/01 10:21:42 +01
00
v442  Deploy 11111111        another.name@email.se       2018/02/27 09:44:47 +01
00
v441  Set SOME_FEATUREFLA…   another.name@email.se       2018/02/27 09:43:15 +01
00
v440  Deploy 11111111        name@email.se               2018/02/25 23:33:54 +01
00
v439  Deploy 11111111        exampleExampleson@email.se  2018/02/19 14:14:28 +01
00
v438  Deploy 11111111        exampleExampleson@email.se  2018/02/19 11:01:23 +01
00
v437  Deploy 11111111        exampleExampleson@email.se  2018/02/16 15:26:33 +01
00
v436  Deploy 11111111        exampleExampleson@email.se  2018/02/16 14:05:52 +01
00
v435  Deploy 11111111        exampleExampleson@email.se  2018/02/16 13:35:59 +01
00
v434  Deploy 11111111        exampleExampleson@email.se  2018/02/16 13:00:55 +01
00
v433  Deploy 11111111        exampleExampleson@email.se  2018/02/15 18:45:37 +01
00
v432  Deploy 11111111        exampleExampleson@email.se  2018/02/15 18:36:40 +01
00
```

After this change, it is displayed correctly as

```
$ heroku releases -a some-app
=== some-app Releases - Current: v446
v446  Deploy 11111111      another.name@email.se       2018/03/07 13:54:13 +0100
v445  Set SOME_FEATUREFL…  name@email.se               2018/03/05 08:09:05 +0100
v444  Deploy 11111111      another.name@email.se       2018/03/01 10:23:18 +0100
v443  Set SOME_FEATUREFL…  another.name@email.se       2018/03/01 10:21:42 +0100
v442  Deploy 11111111      another.name@email.se       2018/02/27 09:44:47 +0100
v441  Set SOME_FEATUREFL…  another.name@email.se       2018/02/27 09:43:15 +0100
v440  Deploy 11111111      name@email.se               2018/02/25 23:33:54 +0100
v439  Deploy 11111111      exampleExampleson@email.se  2018/02/19 14:14:28 +0100
v438  Deploy 11111111      exampleExampleson@email.se  2018/02/19 11:01:23 +0100
v437  Deploy 11111111      exampleExampleson@email.se  2018/02/16 15:26:33 +0100
v436  Deploy 11111111      exampleExampleson@email.se  2018/02/16 14:05:52 +0100
v435  Deploy 11111111      exampleExampleson@email.se  2018/02/16 13:35:59 +0100
v434  Deploy 11111111      exampleExampleson@email.se  2018/02/16 13:00:55 +0100
v433  Deploy 11111111      exampleExampleson@email.se  2018/02/15 18:45:37 +0100
v432  Deploy 11111111      exampleExampleson@email.se  2018/02/15 18:36:40 +0100
```

The most noticeable change is that the descriptions are now truncated 1 character earlier and an extra space before the email is removed -- reducing the width in this example from 82 to 80.

### CODE REVIEW

**Comment by HaraldNordgren** [comment]
I can't quite explain everything that goes on with the length calculations in the current logic. But I found my solution by checking out https://github.com/heroku/cli and manipulating these library files and iterating to get a solution that worked.

For a test like 'shows successful releases' I would expect the mocked terminal to have length 80, but instead I get

    v39  Remove AWS_SECRE…   jeff@heroku.com  2015/11/18 01:36:38 +0000

maybe someone can explain what goes on here. It's consistent with the behavior of the old tests so maybe there is no problem at all.

**Comment by HaraldNordgren** [comment]
The updates to the `yarn.lock` file mainly represents the update of `heroku-cli-util` from v7.0.2 to v.8.0.0. See

```
diff --git a/package.json b/package.json
index fad9e74..a0588f3 100644
--- a/package.json
+++ b/package.json
@@ -23,7 +23,7 @@
     "co": "^4.6.0",
     "co-wait": "^0.0.0",
     "filesize": "^3.5.11",
-    "heroku-cli-util": "https://github.com/HaraldNordgren/heroku-cli-util#options.optimizeKey",
+    "heroku-cli-util": "v8.0.0",
     "inquirer": "^4.0.1",
     "lodash": "^4.17.4",
     "lodash.compact": "^3.0.1",
diff --git a/yarn.lock b/yarn.lock
index f52992b..bcf14da 100644
--- a/yarn.lock
+++ b/yarn.lock
@@ -2067,9 +2067,9 @@ here@0.0.2:
   version "0.0.2"
   resolved "https://registry.yarnpkg.com/here/-/here-0.0.2.tgz#69c1af3f02121f3d8788e02e84dc8b3905d71195"

-"heroku-cli-util@https://github.com/HaraldNordgren/heroku-cli-util#options.optimizeKey":
+heroku-cli-util@v8.0.0:
   version "8.0.0"
-  resolved "https://github.com/HaraldNordgren/heroku-cli-util#da5ee123dd3f18e4dd8c08322ecaefee23fae185"
+  resolved "https://registry.yarnpkg.com/heroku-cli-util/-/heroku-cli-util-8.0.0.tgz#08b86c680aad4a9b81b308edbb877f0be7241f2c"
   dependencies:
     "@heroku-cli/color" "^1.1.3"
     ansi-escapes "^3.0.0"
```

If/when my fix is merged into `heroku-cli-util` I will update the version to v.8.*.

File: test/commands/auth/token.js [hunk 1]
```diff
@@ -9,14 +9,15 @@ const cli = proxyquire('heroku-cli-util', {'netrc-parser': netrc})
...
+const validator = require('validator')
 
 describe('auth:token', () => {
   beforeEach(() => cli.mockConsole())
 
   it('shows logged in user', () => {
     netrc.prototype.machines = {'api.heroku.com': {password: 'myapikey'}}
     return cmd.run({})
-      .then(() => expect(cli.stdout, 'to equal', 'myapikey\n'))
+      .then(() => expect(validator.isUUID(cli.stdout.trim()), 'to be ok'))
```
Comment (HaraldNordgren) [inline test/commands/auth/token.js]: There is definitely something interesting going on here after updating to v8.0.0. It appears to be giving back your own session key.

Not sure if this should be dealt with there though, or in a separate PR.

**Comment by HaraldNordgren** [comment]
Ping @jdxcode @RasPhilCo @ransombriggs

**Comment by jdxcode** [comment]
thanks so much @HaraldNordgren! would you mind just pushing an empty commit or something? I just enabled forked pr building to see if this passes on circle

**Comment by HaraldNordgren** [comment]
Thanks for merging the code! 🙏

Do you know when the next heroku-apps release will be so I can upgrade `cli` to use the new code?

**Pull request merged**